src/solver/solver.cxx

```diff
@@ -1357,6 +1383,140 @@ Field3D Solver::globalIndex(int localStart) {
  * Running user-supplied functions
  **************************************************************************/
 
+int Solver::run_rhs_se(BoutReal t, bool linear) {
+  int status;
+
+  Timer timer("rhs");
+
+  if (first_rhs_call) {
+    // Ensure that nonlinear terms are calculated on first call
+    linear = false;
+    first_rhs_call = false;
+  }
+
```

Comment:
Why does the explicit RHS care about this flag?  I was under the impression that this was added to streamline calculations within the implicit solvers.
```suggestion
```